concent_api/utils/logging.py

```diff
@@ -0,0 +1,71 @@
+from logging                        import getLogger
+
+from golem_messages.message.base    import Message
+
+from utils.helpers                  import get_task_id_from_message
+
+
+logger = getLogger(__name__)
+
+
+def log_message_received(message: Message, client_public_key: str):
+    logger.info('A message has been received in `send/` -- MESSAGE_TYPE: {} -- TASK_ID: {} -- CLIENT PUBLIC KEY: {}'.format(
+        message.TYPE,
+        get_task_id_from_message(message) or '',
+        client_public_key,
+    ))
+    logger.debug('A message has been received in `send/` -- MESSAGE: {}'.format(
+        message,
```

Comment:
Is this enough to print the whole message? Does this include the whole message content, signature and nested messages?

Reply:
Yes, it shows everything, take a look on output:
```
ForceReportComputedTask(header=MessageHeader(type_=4001, timestamp=1510912800, encrypted=False), sig=None, slots=[['task_to_compute', TaskToCompute(header=MessageHeader(type_=2002, timestamp=1510912800, encrypted=False), sig=b'\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00', slots=[['requestor_id', None], ['requestor_public_key', None], ['provider_id', None], ['provider_public_key', None], ['compute_task_def', {'deadline': 1510915020, 'task_id': '1'}], ['package_hash', None], ['concent_enabled', None]])], ['result_hash', None]])
```